source/view/blocks.js

```diff
@@ -1673,12 +1675,12 @@ Blockly.JavaScript['draw_triangle'] = function(block) {
 
   var moveC = constructBlockExeFuncCall( block, actionC, 'moveRadial' );
 
-  var start = "vwf.fireEvent( '" + vwf_view.kernel.application() + 
-                  "', 'blockExecuted', " + " [ 'startTriangle', '" + block.id + "', '" + block.data + "', " + 1 + " ] );\n";
-  var end = "vwf.fireEvent( '" + vwf_view.kernel.application() + 
-                  "', 'blockExecuted', " + " [ 'endTriangle', '" + block.id + "', '" + block.data + "', " + 1 + " ] );\n";
-  var mark = "vwf.fireEvent( '" + vwf_view.kernel.application() + 
-                  "', 'blockExecuted', " + " [ 'markPoint', '" + block.id + "', '" + block.data + "', " + 1 + " ] );\n";
+  var start = "vwf.callMethod( '" + vwf_view.kernel.application() + 
```

Comment:
Yes, this is the Blockly driver. I have tried using vwf_view.kernel with mixed success in the past. This seems to guarantee that the method is called or event fired. I am not sure why.